src/request/selectors/RequestDetailDataSelectors.ts

```diff
@@ -1,19 +1,73 @@
 import { IRequestState } from '../stores/IRequestState';
 
 import { createSelector } from 'reselect';
+import * as _ from 'lodash';
 
+export const getFilterState = (state: IRequestState) => state.detail.data.filters;
 export const getOperations = (state: IRequestState) => state.detail.data.operations;
-export const getSelectedIndex = (state: IRequestState) => state.detail.data.selectedIndex;
+export const getSelectedOperationIdState = (state: IRequestState) => state.detail.data.selectedOperationId;
 
 export const getTotalOperationCount = createSelector(
     getOperations,
     operations => {
         return operations.length;
     });
 
+export const getFilteredOperations = createSelector(
+    getFilterState,
+    getOperations,
+    (filterState, operations) => {
+        const hiddenDatabases = _.keys(_.omitBy(filterState, isShown => isShown));
+        
+        let filteredOperations = [];
+        
+        operations.forEach((operation, index) => {
+            if (!_.includes(hiddenDatabases, operation.database)) {
+                filteredOperations.push({
+                    ordinal: index + 1,
+                    operation: operation
+                })
+            }
+        });
+        
+        return filteredOperations;
+    });
+
+export const getSelectedOperationId = createSelector(
+    getFilteredOperations,
+    getSelectedOperationIdState,
+    (operations, selectedOperationIdState) => {       
+        return selectedOperationIdState === '' && operations.length > 0 
+            ? operations[0].operation.id 
+            : selectedOperationIdState;
+    });
+
 export const getSelectedOperation = createSelector(
+    getFilteredOperations,
+    getSelectedOperationId,
+    (operations, selectedOperationId) => {
+        // TODO: Can this (need this) be optimized by building a map of id --> operation?
+        return _.find(operations, operation => operation.operation.id === selectedOperationId);
+    });
```

Comment:
I actually did a little spike to see whether two mappings helped at all in the case of the Data tab.  To make it work, we'd need to resort the messages after any filter is applied in order to keep a mapping of filtered IDs to items, as we want ensure that only filtered items can be selected.  So it's a choice of a scan per render or a resort per filter change.  Either approach is probably negligibly different, given the total number of operations per typical request.